### PR TITLE
test: add sample network circuit relay tests

### PR DESCRIPTION
**Pull request by jacobheun**
Added some additional tests around circuit relays to make sure that various dials are working properly and that peer connection events are being sent for peers connecting via the relay (https://github.com/libp2p/js-libp2p-circuit/issues/26)

### CODE REVIEW

File: test/circuit-relay.node.js [hunk 1]
```diff
@@ -136,4 +137,189 @@ describe(`circuit`, function () {
...
+      expect(err).to.not.exist()
+
+      bootstrapPeer = infos[0]
+      tcpPeer1 = infos[1]
+      tcpPeer2 = infos[2]
+      wsPeer1 = infos[3]
+      wsPeer2 = infos[4]
+
+      // Setup the addresses of our nodes
+      bootstrapPeer.multiaddrs.add('/ip4/127.0.0.1/tcp/9100')
```
Comment (dryajov) [inline test/circuit-relay.node.js]: should use ephemeral ips and ports

**Review by dryajov — CHANGES_REQUESTED** [review]
should use ephemeral ips and ports, other than that 👍

**Comment by jacobheun** [comment]
Ports are now being discovered. I noticed I was also not stopping the swarms after the tests, so that is being done now. Most of the tests in switch are not using ephemeral ports. If we're doing actual connection testing we should migrate those over.

**Comment by dryajov** [comment]
> Most of the tests in switch are not using ephemeral ports.

huh, true - I forgot about #227! But you're not using port `0` either way so I think this will work?

**Review by dryajov — APPROVED** [review]

**Pull request merged**